appservice/src/deploy/deployZip.ts

```diff
@@ -44,6 +44,16 @@ export async function deployZip(client: SiteClient, fsPath: string, configuratio
             const kuduClient: KuduClient = await getKuduClient(client);
             await kuduClient.pushDeployment.zipPushDeploy(fs.createReadStream(zipFilePath), { isAsync: true });
             await waitForDeploymentToComplete(client, kuduClient);
+            if (asp && asp.sku && asp.sku.tier && asp.sku.tier.toLowerCase() === 'basic') {
+                const deployments: number = (await kuduClient.deployment.getDeployResults()).length;
+                // if this is the first deployment, implement a 10 second delay for apps in a basic plan due to long start times
+                if (deployments === 1) {
+                    const delay: (delayMs: number) => Promise<void> = async (delayMs: number): Promise<void> => {
```

Comment:
Would be more readable as a nested or (better) file-level function

Reply:
Also, I thought there was supposed to be a message?

Reply:
I think Stephen is talking about "delay", but honestly I'd like all of this new code in it's own function. Just so that it looks cleaner and so that it'll be easier to remove once this workaround isn't necessary